internal/charmstore/search.go

```diff
@@ -98,11 +92,59 @@ func (s *Store) UpdateSearch(r *router.ResolvedURL) error {
 		}
 		return errgo.Notef(err, "cannot get %s", r)
 	}
-	baseEntity, err := s.FindBaseEntity(entity.BaseURL, "acls")
+	baseEntity, err := s.FindBaseEntity(entity.BaseURL)
 	if err != nil {
-		return errgo.Mask(err, errgo.Is(params.ErrNotFound))
+		return errgo.Notef(err, "cannot get %s", entity.BaseURL)
 	}
-	doc, err := s.searchDocFromEntity(&entity, baseEntity)
+	if err := s.updateSearchEntity(&entity, baseEntity); err != nil {
+		return errgo.Notef(err, "cannot update search record for %q", entity.URL)
+	}
+	return nil
+}
+
+// UpdateSearchBaseURL updates the search record for all entities with the specified base URL.
```

Comment:
// It must be called whenever the entry for the given URL in
// the BaseEntitites collection has changed.
?